source/MaterialXTest/ShaderValid.cpp

```diff
@@ -371,6 +371,51 @@ static mx::OslValidatorPtr createOSLValidator(bool& orthographicView, std::ostre
 }
 #endif
 
+//
+// Shader validation options structure
+//
+class ShaderValid_TestOptions
+{
+public:
+    // Filter list of files to only run validation on.
+    MaterialX::StringVec overrideFiles;
+    // Set to true to always dump glsl generated files to disk
+    bool dumpGlslFiles = false;
+    // Execute GLSL tests
+    bool runGLSLTests = true;
+    // Execute OSL tests
+    bool runOSLTests = true;
+    // Run using a set of interfaces:
+    // - 3 = run complete + reduced.
+    // - 2 = run complete only (default)
+    // - 1 = run reduced only.
+    int shaderInterfaces = 2;
+    // Non-shader GLSL geometry file
+    MaterialX::FilePath glslNonShaderGeom = "/sphere.obj";
+    // Shader GLSL geometry file
+    MaterialX::FilePath glslShaderGeom = "/shaderball.obj";
+    // Add more options as desired...
+};
+
+// Create a list of generation options based on unit test options
+void getGenerationOptions(std::vector<mx::GenOptions>& optionsList, const ShaderValid_TestOptions& testOptions)
```

Comment:
Minor: Coding style - output variable should come last in the function signature.